lib/src/components/Divider/Divider.tsx

```diff
@@ -0,0 +1,71 @@
+import { assignInlineVars } from "@vanilla-extract/dynamic";
+import {
+  dividerContainerHorizontal,
+  dividerContainerVertical,
+  dividerHorizontal,
+  dividerThemeVar,
+  dividerVertical,
+  labelHorizontal,
+  labelVertical,
+} from "./divider.css";
+import { DividerProps } from "./divider.types";
+
+export const Divider = ({
+  type = "solid",
+  label,
+  labelPosition = "center",
+  color = "grey",
+  labelColor = "#000",
+  labelBackground = "#fff",
+  orientation = "horizontal",
+  size = "1px",
+  ...nativeProps
+}: DividerProps) => {
+  const dividerClass =
+    orientation === "horizontal"
+      ? dividerHorizontal({
+          type,
+        })
+      : dividerVertical({
+          type,
+        });
+  const labelClass =
+    orientation === "horizontal"
+      ? labelHorizontal({
+          labelPosition,
+        })
+      : labelVertical({
+          labelPosition,
+        });
+
+  return (
+    <div
+      className={`divider-container ${
+        orientation === "horizontal"
+          ? dividerContainerHorizontal
+          : ` ${dividerContainerVertical}`
+      }`}
+    >
+      <div
+        className={`divider ${dividerClass} `}
```

Comment:
what is the `divider` className here? do not put simple classnames like these as they may clash with consuming project. always put classNames from vanilla-extract as they are unique

Reply:
in order for user to customize or override any default styles...he can simply use these classNames to customize

Reply:
not really an elegant way of making people use our classnames, will not be helpful if people want to use multiple styles or something like tailwind. rather use native classname and put it beside our classes so they can customize it

eg `className={${dividerClass} ${nativeClassName}}`

gives freedom to user to give classname and wont pollute the dom, dont want the divider class to pollute all other instances of dividers in an app

Reply:
okay